src/components/ui/swapContainer.js

```diff
@@ -98,51 +108,105 @@ const SwapContainer = ({ token, header }) => {
   const [balToken, setBalance] = useState("0");
   useEffect(() => {
     (async () => {
-      if (authContext.activeUser && token.name === "IQ") {
+      if (
+        token.chain === "EOS" &&
+        authContext.activeUser &&
+        token.name === "IQ"
+      ) {
         setBalance(await getUserTokenBalance(authContext));
-      } else if (wallet.account && token.name === "pIQ") {
+      } else if (
+        token.chain === "Ethereum" &&
+        wallet.account &&
+        token.name === "pIQ"
+      ) {
         setBalance(await getPTokensUserBalance(wallet));
+      } else if (
+        token.chain === "Ethereum" &&
+        wallet.account &&
+        token.name === "IQ"
+      ) {
+        const balance = Number(await getTokensUserBalance(wallet));
+        setBalance(balance);
+        setParentBalance(balance);
       }
     })();
   }, [authContext, wallet, token]);
 
+  const handleTriggerFillInput = () => {
+    swapRef.current.value = balToken;
+    setFilled(swapRef.current.value);
+  };
+
+  const handleOnInputChange = event => {
+    let { value } = event.target;
+    const { min, max } = event.target;
+    value = Math.max(Number(min), Math.min(Number(max), Number(value)));
+    swapRef.current.value = value;
+    setFilled(value);
+  };
+
   return (
     <SwapContainerWrapper>
       <SwapTokenHeader className="text-capitalize">
-        <SwapBalance>{`${t("balance")}: ${balToken}`}</SwapBalance>
+        <SwapBalance>
+          <ClickToFillBtn
+            size="sm"
+            variant="transparent"
+            onClick={handleTriggerFillInput}
+          >
+            {`${t("balance")}: ${balToken}`}
+          </ClickToFillBtn>
+        </SwapBalance>
         <SwapHeader>{t(header.toLowerCase())}</SwapHeader>
       </SwapTokenHeader>
-      <SwapTokenContainer>
-        <div>
+      <Row className="d-flex flex-row justify-content-between">
+        <Col xs={9} md={9} lg={9}>
           <SwapTokenInput
-            autoComplete="off"
+            type="number"
+            min={0}
+            max={balToken}
+            disabled={wallet.account === null}
             name={`${header}Amount`}
             placeholder={token ? `0.${"0".repeat(token.precision)}` : "0.000"}
+            onChange={handleOnInputChange}
```

Comment:
this breaks the EOS Bridge https://github.com/EveripediaNetwork/iq-bridge-ui/issues/33